src/support/z_cowboy_log_stream_handler.erl

```diff
@@ -0,0 +1,39 @@
+-module(z_cowboy_log_stream_handler).
+-author("David de Boer <david@ddeboer.nl>").
+
+-behavior(cowboy_stream).
+
+-export([
+    init/3,
+    data/4,
+    info/3,
+    terminate/3
+]).
+
+-type state() :: any().
+
+-spec init(cowboy_stream:streamid(), cowboy_req:req(), cowboy:opts())
+	-> {cowboy_stream:commands(), {module(), state()} | undefined}.
+init(StreamID, Req, Opts) ->
+    z:debug_msg(?MODULE, ?LINE, Req),   %% Map with request headers and properties
+    cowboy_stream:init(StreamID, Req, Opts).
+
+-spec data(cowboy_stream:streamid(), cowboy_stream:fin(), binary(), {Handler, State} | undefined)
+	-> {cowboy_stream:commands(), {Handler, State} | undefined}
+	when Handler::module(), State::state().
+data(StreamID, IsFin, Data, {Handler, State0}) ->
+    z:debug_msg(?MODULE, ?LINE, Data),  %% Response body
+    cowboy_stream:data(StreamID, IsFin, Data, {Handler, State0}).
```

Comment:
> Should stream handlers call cowboy_stream? That sounds weird.

This was changed recently in Cowboy 2 and is the right way to go about it.

> For logging with information on the zotonic context you need the responses returned by cowboy_stream calls.

I have those responses (see code above) but they only contain the HTTP message, *not* any (Zotonic) context.